src/model/attributes/relations/MorphTo.ts

```diff
@@ -101,40 +102,36 @@ export class MorphTo extends Relation {
   }
 
   /**
-   * Since we do not know the child model ahead of time, we cannot add any
+   * Add eager constraints. Since we do not know the related model ahead of time, we cannot add any
    * eager constraints.
    */
   addEagerConstraints(_query: Query, _models: Collection): void {
     return
   }
 
-  /**
-   * Execute the eager loading query.
-   */
-  getEager(_query: Query): Collection {
-    return []
-  }
-
   /**
    * Find and attach related children to their respective parents.
    */
-  match(
-    relation: string,
-    models: Collection,
-    _results: Collection,
-    query: Query
-  ): void {
-    const queries = this.createRelatedQueries(query)
-
-    models.forEach((model) => {
-      const type = model[this.morphType]
-      const id = model[this.morphId]
-
-      const related =
-        type !== null && id !== null ? queries[type].find(id) : null
-
-      model.$setRelation(relation, related)
-    })
+  match(relation: string, models: Collection, query: Query): void {
+    // Create dictionary that contains relationships.
+    const dictionary = this.buildDictionary(query, models)
+
+    if (Object.keys(dictionary).length < 1) {
+      // Set relations to null if dictionary is empty
+      models.forEach((model) => {
+        model.$setRelation(relation, null)
+      })
+    } else {
+      // Otherwise, set model relations using the dictionary
+      models.forEach((model) => {
+        const type = model[this.morphType]
+        const id = model[this.morphId]
+
+        const related = dictionary[type]?.[id] ?? null
+
+        model.$setRelation(relation, related)
+      })
+    }
```

Comment:
I don't think we need this if/else statement. Seems redundant as the relation will be null if it's not indexed in the dictionary. Let's keep it simple and use one loop branch here. What do you think?

Reply:
I wasn't 100% sure if we wanted to favour the simpler approach and opt'd to attempt to optimize a bit. Personally, I prefer the simpler approach 👍

Reply:
Awesome 🎉